components/brave_wallet/browser/zcash/zcash_wallet_service.cc

```diff
@@ -321,11 +340,111 @@ void ZCashWalletService::GetUtxos(const std::string& chain_id,
   }
 }
 
+bool ZCashWalletService::SignTransactionInternal(
+    ZCashTransaction& tx,
+    const mojom::AccountIdPtr& account_id) {
+  auto addresses = keyring_service_->GetZCashAddresses(*account_id);
+  if (!addresses || addresses->empty()) {
+    return false;
+  }
+
+  std::map<std::string, mojom::ZCashKeyIdPtr> address_map;
+  for (auto& addr : *addresses) {
+    address_map.emplace(std::move(addr.first), std::move(addr.second));
+  }
+
+  for (size_t input_index = 0; input_index < tx.inputs().size();
+       ++input_index) {
+    auto& input = tx.inputs()[input_index];
+
+    if (!address_map.contains(input.utxo_address)) {
+      return false;
+    }
+    auto& key_id = address_map.at(input.utxo_address);
+
+    auto pubkey = keyring_service_->GetZCashPubKey(account_id, key_id);
+    if (!pubkey) {
+      return false;
+    }
+
+    auto signature_digest =
+        ZCashSerializer::CalculateSignatureDigest(tx, input);
+
+    if (signature_digest.size() != 32) {
+      return false;
+    }
```

Comment:
CalculateSignatureDigest should return std::array of 32 bytes to avoid such checks